R/nba_scraper.R

```diff
@@ -1,26 +1,154 @@
 #' Scrape NBA statistics data from ESPN into CSV file
 #'
-#' Scrape the tabular data from ESPN NBA website into a csv file.
+#' Scrape the tabular data from ESPN NBA website into a csv file using RSelenium.
 #'
-#' @param season_year string
+#' @param season_year int from 2001 to 2019 (upper limit based on latest year)
 #' @param season_type string
 #' @param csv_path string
+#' @param port int with L suffix
+#' @param sel_browser string
 #' @param nba_data_env_name string
 #'
 #' @export
+#' @importFrom magrittr %>%
+#' @importFrom utils write.csv
+#' @importFrom rvest html_nodes
+#' @importFrom rvest html_text
 #'
 #' @examples
-#' # Regular season for 2018/2019 while assigning it into global env as an object called nba_2018
-#' nba_scraper(season_year = "2018", season_type = "regular",
-#'             csv_path = "nba_2018_reg.csv", nba_data_env_name = "nba_2018")
-#' # Playoff season for 2016/2017 without assigning it into global env as an object
-#' nba_scraper(season_year = "2016", season_type = "playoff",
-#'             csv_path = "nba_2018_reg.csv",nba_data_env_name = NULL)
-nba_scraper <- function(season_year, season_type, csv_path, nba_data_env_name = NULL) {
-
-  # If nba_data_env_name is given, create a dataframe by reading it
-  #if (!is.null(nba_data_env_name)) {
-  #  assign(nba_data_env_name, read.csv(csv_path), envir = .GlobalEnv)
-  #}
-  print("Data scraping completed")
+#' # Scrape regular season 2018/19 using "chrome" driver and
+#' # create "nba_2018_regular" tibble in Global Env
+#' \dontrun{
+#' #' nba_scraper(2018, season_type = "regular", "nba_2018.csv",
+#'             port=4445L, sel_browser = "chrome",
+#'             nba_data_env_name = "nba_2018_regular")
+#' }
+#'
+#' # Scrape playoffs season 2017/18 using "firefox" driver without
+#' # creating tibble in Global Env
+#' \dontrun{
+#' nba_scraper(2017, season_type = "playoffs", "nba_2017_playoffs.csv",
```

Comment:
I think the name must be alligned to what we see in the webpage
```suggestion
#' nba_scraper(2017, season_type = "postseason", "nba_2017_postseason.csv",
```

Reply:
this has been changed accordingly with a more recent commit!